posix: implement sched_rr_get_interval()

moonlight83340 opened the pull request:
This is part of the See https://github.com/zephyrproject-rtos/zephyr/issues/51211 (RFC https://github.com/zephyrproject-rtos/zephyr/issues/51211).

sched_rr_get_interval() is required as part of _POSIX_PRIORITY_SCHEDULING Option Group.

For more information, please refer to https://pubs.opengroup.org/onlinepubs/9699919799/functions/sched_rr_get_interval.html

Function is actually placeholders and just return `ENOSYS`
since Zephyr does not yet support processes or process scheduling.

Fixes #66961

moonlight83340 committed:
doc: posix: mark sched_rr_get_interval as supported

`sched_rr_get_interval()` is now implemented,mark it so.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

moonlight83340 committed:
posix: sched: Implement tests for sched_rr_get_interval

Implement tests for `sched_rr_get_interval()` .
Function is actually placeholders and just return `ENOSYS`
since Zephyr does not yet support processes or process scheduling.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

ycsin commented on include/zephyr/posix/sched.h:
```diff
@@ -10,6 +10,8 @@
 
 #include "posix_types.h"
 
+#include "time.h"
```
I think this should be grouped with the `posix_types.h` above

moonlight83340 commented:
I will rebase tonight, my other PR has been merged so maybe the conflic is from that.

cfriedt commented on include/zephyr/posix/sched.h:
```diff
@@ -9,6 +9,7 @@
 #include <zephyr/kernel.h>
 
 #include "posix_types.h"
+#include "time.h"
```
This is a standard ISO C header file, so it should have angle brackets and have a newline after private headers (sorry, this is not yet built into .clang-format)


```suggestion

#include <time.h>
```

moonlight83340 replied:
Noted, thank you !

cfriedt reviewed (commented):
Just a minor include nit, otherwise LGTM

moonlight83340 committed:
posix: sched: Implement sched_rr_get_interval

Implement `sched_rr_get_interval()` POSIX APIs
as a part of PSE53 `_POSIX_PRIORITY_SCHEDULING` option group.
Functions is actually placeholders and just return `ENOSYS`
since Zephyr does not yet support processes or process scheduling.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

moonlight83340 commented:
Rebase.
